docs/source/installation.rst

```diff
@@ -211,3 +211,17 @@ These dependencies are managed, installed and referenced by the library using `P
 No additional dependencies should be needed after running the pip install.
 
 .. _Poetry: https://python-poetry.org/docs/
+
+Version compatibility
+=====================
+
+The Mesh Python SDK will perform the version compatibility check when connecting to the Mesh server.
```

Comment:
I'd still state that for `aio` we do the compatibility check when creating session.